wp/lib/bap_wp/src/precondition.ml

```diff
@@ -728,51 +728,54 @@ let mk_env
   Env.mk_env ~subs ~specs ~default_spec ~indirect_spec ~jmp_spec ~int_spec ~exp_conds ~num_loop_unroll
     ~arch ~freshen_vars ~use_fun_input_regs ~stack_range ~data_section_range ctx var_gen
 
+let conditional_jmp (jmp : Jmp.t) (env : Env.t) (target_pre : Constr.t)
+    (post : Constr.t) : Constr.t * Env.t =
+  let ctx = Env.get_context env in
+  let cond = Jmp.cond jmp in
+  let cond_val, hooks, env = exp_to_z3 cond env in
+  debug "\n\nJump when %s:\n%s\n%!"
+    (Expr.to_string cond_val) (hooks_to_string hooks);
+  let cond_size = BV.get_size (Expr.get_sort cond_val) in
+  let false_cond = Bool.mk_eq ctx cond_val (z3_expr_zero ctx cond_size) in
+  let is_unconditional =
+    match cond with
+    | Bil.Types.Int w -> Word.is_one w
+    | _ -> false
+  in
+  let ite =
+    if is_unconditional then
```

Comment:
In `conditional_call/run_wp.sh`, we test a situation when `ite` is assigned `Constr.mk_ite jmp (Bool.mk_not ctx false_cond) target_pre post`, since we create `if not(= (ZF0) 0) then ...`. However, if I understand this code correctly, we don't test the case when `ite` is assigned a value based on `is_unconditional`. Would it be worth adding a test case for this `is_unconditional` case?

Reply:
Generally, it's a pretty good idea to try to test all code paths. I don't have particularly strong feelings about this specific case though. We're already pretty delinquent in terms of coverage.

Reply:
I can definitely add one!

Reply:
With `sample_binaries/unconditional_call` added, we now cover the two conditions of `ite` inside `conditional_jmp`! Thanks!